benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/MatcherSelectCandidatesAzureBenchmark.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    // Generated from https://github.com/Azure/azure-rest-api-specs
+    public partial class MatcherSelectCandidatesAzureBenchmark : MatcherBenchmarkBase
+    {
+        private const int SampleCount = 100;
+
+        private BarebonesMatcher _baseline;
+        private DfaMatcher _dfa;
+
+        private int[] _samples;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            SetupEndpoints();
+
+            SetupRequests();
+
+            // The perf is kinda slow for these benchmarks, so we do some sampling
+            // of the request data.
+            _samples = SampleRequests(EndpointCount, SampleCount);
+
+            _baseline = (BarebonesMatcher)SetupMatcher(new BarebonesMatcherBuilder());
+            _dfa = (DfaMatcher)SetupMatcher(new DfaMatcherBuilder());
+        }
+
+        [Benchmark(Baseline = true, OperationsPerInvoke = SampleCount)]
+        public unsafe void Baseline()
+        {
+            for (var i = 0; i < SampleCount; i++)
+            {
+                var sample = _samples[i];
+                var httpContext = _requests[sample];
```

Comment:
Can `_requests` and `_endpoints` be protected instead of internal? Also IMO rename to `Requests` and `Endpoints`. I reserve underscores for only private fields.

Reply:
The can be `private protected`. The reason they weren't is that benchmark classes have to be public, and many of these types are internal.

Reply:
BTW it take a really really long time to rename these due to the generated code 🤣